etc/kayobe/kolla.yml

```diff
@@ -250,12 +250,26 @@ kolla_build_blocks:
         sed -i -e '/\[{{ repo.tag }}\]/,/^\[/ s/^\(mirrorlist *=.*\)/#\1/g' \
                -e '/\[{{ repo.tag }}\]/,/^\[/ s/^[# ]*\(baseurl *=.*\)/#\1/g' \
                -e '/\[{{ repo.tag }}\]/,/^\[/ s/^[# ]*\(metalink *=.*\)/#\1/g' \
+    {% if stackhpc_repo_mirror_username is truthy %}
+               -e '/\[{{ repo.tag }}\]/,/^\[/ s|^\(name.*\)|\1\nusername={{ stackhpc_repo_mirror_username }}|' \
+               -e '/\[{{ repo.tag }}\]/,/^\[/ s|^\(name.*\)|\1\npassword={{ stackhpc_repo_mirror_password }}|' \
```

Comment:
I'm not sure if there's a good way around it, but this could create some hard-to-debug errors if the password contains a `|`

Reply:
We could escape it, but a fix should go to stackhpc/yoga. @jovial had some valid concerns around the passwords being in lower layers of the images, even though we remove them, so we may need to revisit this.